acra-censor/handlers/querycapture_handler.go

```diff
@@ -192,25 +148,112 @@ func (handler *QueryCaptureHandler) GetForbiddenQueries() []string {
 	}
 	return forbiddenQueries
 }
+
 func (handler *QueryCaptureHandler) SetSerializationTimeout(timeout time.Duration) {
 	handler.serializationTimeout = timeout
 }
+
 func (handler *QueryCaptureHandler) GetSerializationTimeout() time.Duration {
 	return handler.serializationTimeout
 }
 
-func (handler *QueryCaptureHandler) Serialize() error {
-	jsonFile, err := json.Marshal(handler.Queries)
+func (handler *QueryCaptureHandler) DumpAllQueriesToFile() error {
+	// open or create file, NO APPEND
+	f, err := os.OpenFile(handler.filePath, os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil {
+		log.WithError(ErrCantOpenFileError).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCensorIOError)
+		return err
+	}
+
+	// write all queries
+	return AppendQueries(handler.Queries, f)
+}
+
+func (handler *QueryCaptureHandler) DumpBufferedQueriesToFile(openedFile *os.File) error {
+	// write buffered queries
+	buf := handler.BufferedQueries
+
+	// nothing to dump
+	if len(buf) == 0 {
+		return nil
+	}
+
+
+	// clean buffered queries
+	handler.BufferedQueries = nil
+	err := AppendQueries(buf, openedFile)
 	if err != nil {
 		return err
 	}
-	return ioutil.WriteFile(handler.filePath, jsonFile, 0600)
+
+	return nil
 }
-func (handler *QueryCaptureHandler) Deserialize() error {
-	var bufferBytes []byte
-	bufferBytes, err := ioutil.ReadFile(handler.filePath)
+
+func (handler *QueryCaptureHandler) ReadAllQueriesFromFile() error {
+	q, err := ReadQueries(handler.filePath)
 	if err != nil {
+		log.WithError(ErrCantReadQueriesFromFileError).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCensorIOError)
 		return err
 	}
-	return json.Unmarshal(bufferBytes, &handler.Queries)
+
+	// read existing queries from file
+	handler.Queries = q
+	return nil
 }
+
+func AppendQueries(queries []QueryInfo, openedFile *os.File) error {
+	if len(queries) == 0 {
+		return nil
+	}
+
+	lines, err := SerializeQueries(queries)
+	if err != nil {
+		return err
+	}
+
+	if _, err := openedFile.Write(lines); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+
+func SerializeQueries(queries []QueryInfo) ([]byte, error) {
+	var linesToAppend []byte
+	for _, queryInfo := range queries {
+		jsonQueryInfo, err := json.Marshal(queryInfo)
+		if err != nil {
+			return nil, err
+		}
+		if len(jsonQueryInfo) > 0 {
+			jsonQueryInfo = append(jsonQueryInfo, "\n"...)
```

Comment:
can we change to `append(jsonQueryInfo, '\n')` (single quotes are one character literal, double quotes are string literal that iterable)? after `"\n"... I think about _"is it only 1 character or it 2 characters and therefore we use ... ?"_ )